content/en/contribute/code/design-docs.md

```diff
@@ -0,0 +1,67 @@
+---
+title: "Design Documents Guide"
+linkTitle: "Design Docs"
+weight: 12
+description: >
+  Guidelines for writing technical design documents
+---
+
+## What are design docs?
+Software development is not just about writing code, but rather about solving problems and building the right solutions. Before diving into an initiative or feature and start coding, it’s essential that the developers (and other team members) have a high-level understanding of what a solution might look like. 
+
+Design docs are informal documents that the leading developer of a certain piece of software create before they start the actual coding of a solution. The design doc contains high-level technical design decisions and alternatives that were considered when making those decisions. 
+
+## Why write design docs?
+
+Besides being an excellent way of documenting software design, design docs come with several additional benefits:
+
+* Involve the team early and get feedback before the implementation, which is more costly.
+* Identify solution concerns and issues rapidly, as making changes to a solution in the design phase is faster.
+* Provide a fantastic way to document technical decisions, which benefits the team, the community, and future contributors.  
+* Ensure the consideration of alternative solutions, assumptions and eventual constraints.
+* Achieve consensus in the team around the design and get everyone on the same page.
+
+## What could a design doc contain?
+
+The design doc can be added to the GitHub issue related to the feature or initiative to be implemented or in a Google Doc. 
+
+The list below contains a non-exhaustive list of items a design doc could cover.
+
+#### Context and scope
+An overview of the context in which the piece of software is being built and what is actually being built. It's important to keep this section succint as it's only meant to bring the readers up to speed with the background facts.
+
+#### Goals and non-goals
+A short list of what the goals of the piece of software are, and, very importantly, what the non-goals are. The non-goals aren’t negated goals such as "The app shouldn’t crash", but rather things that could be goals, but are explicitly chosen not to be goals, as for example "FHIR compliace of the API". Please note that a solution could cover a non-goal, as long as it doesn’t introduce trade-offs that prevent achieving the goals.
```

Comment:
I prefer "out of scope" to "non-goals" and I think it makes the section easier to understand.

```suggestion
#### Goals and scope
A short list of what the goals of the piece of software are, and, very importantly, what is out of scope. The out of scope items are explicitly chosen not to be goals, as for example "FHIR compliace of the API". Please note that a solution could cover out of scope items, as long as it doesn’t introduce trade-offs that prevent achieving the goals.
```